### PR TITLE
PresentationController demo

### PR DESCRIPTION
**Pull request by tbaranes**
Related to #281 
No screenshots (the view is too big), check out the branch! It's as we said: sliders, pickers... everything is customisable before looking at our result!
Next step: documentation (😨)

**Note:** I didn't let the `Custom` size / position, I think it's overkill to have a `Custom` in a picker.

### CODE REVIEW

**Comment by JakeLin** [comment]
@tbaranes thanks, I like the demo especially the color slider 👍.

I have a few comments
1. Have changed the back button and the segue to make it consistent as the other demo.
2. Shall we change the "Done" to "Present"? You can make the call 😉
3. The enums like `animations`, `keyboardTranslations` and `blurEffectStyles`. Shall we change the empty ("") to something more obviously like "none", "not set" or "N/A"? Because it is hard for me to find "" in the picker view. In swfit3, we will have `.none` for all custom enums. Maybe we can use "none".
4. `Opacity` and `Shadow` seem not working 😢

**Comment by tbaranes** [comment]
Thanks for the feedback @JakeLin! I changed to reflect 2 and 3, I agree with you, that's better.
For the `opacity`, and `shadow`, I have to take time to look into it, will do it later, but nice catch 👍

**Comment by JakeLin** [comment]
@tbaranes cool.

**Comment by lastMove** [comment]
Very nice Demo !! 
Well done, It 's fun to use :) 
I was a little bit confused because of the "Present" button. It can be useful to add another present button at the end of the scroll (it's totally subjective). 
Great job.

**Comment by tbaranes** [comment]
That's a good idea @lastMove! I just added it, a new button is waiting for you at the bottom of the scrollview. I didn't have that much ideas for the design, so it's an orange button 💩 (in fact, it's not that bad... but if you have preferences, let me know)

**Comment by tbaranes** [comment]
@JakeLin The opacity bug was just an error in the setup (the value wasn't set), but I have a doubt about the implementation of that one, in order to make it work, I apply the `opacity` directly on the `backgroundColor` when setting `fillColor` instead of using `opacity`. When using `opacity`, the alpha is always 0.

Should it be that way? 🤔

About the shadow, I take a look into it, I have no idea why is not working. All the values are well-settled, and the layer seems good, so I will go back on it later, but if anyone has an idea... 😨

File: IBAnimatableApp/PresentingViewController.swift [hunk 1]
```diff
@@ -0,0 +1,279 @@
...
+    presentedViewController.shadowRadius = CGFloat(sliderShadowRadius.value)
+    presentedViewController.shadowOffset = CGPoint(x: CGFloat(sliderShadowOffsetX.value), y: CGFloat(sliderShadowOffsetY.value))
+
+    setDismissalAnimationTypeIfNeeded(presentedViewController)
+  }
+
+  private func setDismissalAnimationTypeIfNeeded(viewController: AnimatableModalViewController) {
+    // FIXME: Dirty hack to make `Flip` and `CrossDissolve` work properly for dismissal transition.
+    // If we don't apply this hack, both the dismissal transitions of `Flip` and `CrossDissolve` will slide down the modal not flip or crossDissolve(fade).
+    if viewController.presentationAnimationType == "Flip" {
```
Comment (JakeLin) [inline IBAnimatableApp/PresentingViewController.swift:96]: Shall we check dismissalAnimationType is nil here?

Comment (tbaranes) [inline IBAnimatableApp/PresentingViewController.swift:96]: I don't think, if it's nil, the condition won't be fulfilled

**Comment by JakeLin** [comment]
I think it's ok to set the opacity in background color because the dimming view either have a color or blur effect view. And the blur effect has its own opacity.

File: IBAnimatableApp/PresentingViewController.swift (hunk 1 above)
Comment (JakeLin) [inline IBAnimatableApp/PresentingViewController.swift:96]: If it's not nil, shall we use the specified value for dismissalAnimationType?

Comment (tbaranes) [inline IBAnimatableApp/PresentingViewController.swift:96]: Aaaah, sorry I misunderstood, you are right, we have to do the setup fix only if it's nil

Comment (JakeLin) [inline IBAnimatableApp/PresentingViewController.swift:96]: @tbaranes No worries, I should make it more specific, I was typing on my lovely 📱

**Comment by tbaranes** [comment]
Here some news of the shadow:
-  Updating the default values to have a shadow by default if we set a color
- Updating the shadow implementation in order to make it work (and avoid useless if)

I have one last issue: the color from the slider seems to be invalid, if you hardcode the color, the shadow will appear as expected, but not when using the custom one. Well, almost there 👯

**Comment by tbaranes** [comment]
I'm a bit disappointing about this hack, in these lasts commits, the shadow has been fixed. I also change the default value of the presentation in order to have a default shadow just by setting a color.

The main bug was due to the picker color, I don't know why, but the layer didn't process the hex color from the picker. So I replaced each hex by an `UIColor`, that make the shadow work as expected but the sliders vs the result can be weird (their not enough `UIColor`). Ideally, we should make a new sliders and respect the `UIColor`, maybe later?

Anyway, you can review it one more time, I think that time, it's ready to go :)

File: IBAnimatable/AnimatableModalViewController.swift [hunk 2]
```diff
@@ -104,7 +104,7 @@ public class AnimatableModalViewController: UIViewController, PresentationDesign
     }
   }
 
-  @IBInspectable public var shadowOffset: CGPoint = CGPoint(x: CGFloat.NaN, y: CGFloat.NaN) {
+  @IBInspectable public var shadowOffset: CGPoint = CGPoint.zero {
```
Comment (JakeLin) [inline IBAnimatable/AnimatableModalViewController.swift]: We can remove `CGPoint` and just use `.zero`

**Comment by JakeLin** [comment]
👍  well done @tbaranes

**Pull request merged**